API/API/Controllers/ProductController.cs

```diff
@@ -1,109 +1,110 @@
+[ApiController]
+public class ProductController : ControllerBase
 {
     /// <summary>
-    /// Products controllers
+    /// Manage product busines logic
     /// </summary>
-    [Route("/Products")]
-    [ApiController]
-    public class ProductController:ControllerBase
-    {
-        private readonly ProductContext _context;
+    private readonly IProductService _productService;
+
+    /// <summary>
+    /// Constructor
+    /// </summary>
+    /// <param name="productService"></param>
+    public ProductController(IProductService productService) => _productService = productService;
 
-        public ProductController(ProductContext context)
+    /// <summary>
+    /// Get product data
+    /// </summary>
+    /// <param name="idProduct"> product Id </param>
+    /// <returns> product </returns>
+    [HttpGet("{idProduct}")]
+    [ProducesResponseType(StatusCodes.Status200OK, Type = typeof(ProductModel))]
+    [ProducesResponseType(StatusCodes.Status400BadRequest)]
+    public async Task<ActionResult<ProductModel>> GetProduct(int idProduct)
+    {
+        try
         {
-            _context = context;
+            ProductModel product = await _productService.GetProduct(idProduct);
+            return Ok(product);
         }
-
-        /// <summary>
-        /// Get product data
-        /// </summary>
-        [HttpGet]
-        public async Task<ActionResult<ProductItem>> GetProductItem(int id)
+        catch (ExceptionClass ex)
         {
-            var productItem = await _context.ProductItems.FindAsync(id);
-
-            if (productItem == null)
-            {
-                return NotFound();
-            }
-            return productItem;
+            return BadRequest(ex.Message);
         }
+    }
 
-        /// <summary>
-        /// Post product data
-        /// </summary>
-        [HttpPost]
-        public async Task<ActionResult<ProductItem>> PostProductItem(ProductItem productItem)
+    /// <summary>
+    /// Post product data
+    /// </summary>
+    /// <param name="Product"> product </param>
+    /// <returns> product </returns>
+    [HttpPost("{idProduct}")]
+    [ProducesResponseType(StatusCodes.Status200OK, Type = typeof(ProductModel))]
+    [ProducesResponseType(StatusCodes.Status400BadRequest)]
+    public async Task<ActionResult<ProductModel>> PostProduct(int idProduct, [FromBody] ProductModelInput product)
+    {
+        try
         {
-
-            _context.ProductItems.Add(productItem);
-            await _context.SaveChangesAsync();
-
-            return CreatedAtAction(nameof(GetProductItem), new { id = productItem.Id }, productItem);
+            return Ok(await _productService.CreateProduct(new ProductModel(idProduct, product.ProductName, product.ProductPrice)));
         }
-
-        /// <summary>
-        /// Put product data
-        /// </summary>
-        [HttpPut]
-        public async Task<IActionResult> PutProductItem(int id,ProductItem productItem)
+        catch (ExceptionClass ex)
```

Comment:
You must add general class exception, because if you have other exception than ExceptionClass the project will crash

Reply:
Add this in all controllers

Reply:
Done